rest-api/src/main/java/com/cloudbees/workflow/rest/external/RunExt.java

```diff
@@ -407,7 +421,20 @@ private void initStatus(WorkflowRun run) {
         } else if (execution.getCauseOfFailure() != null) {
             setStatus(StatusExt.valueOf(execution.getCauseOfFailure()));
         } else if (execution.isComplete()) {
-            setStatus(StatusExt.SUCCESS);
+            Result r = run.getResult();
+            if (r == Result.NOT_BUILT) {
```

Comment:
I'm not proposing anything - merely stating that `Result` is not final - and normally comparing them would be a bug.
for instance if a plkugin introduced a different Result - lets call it "Whacky" and a pipeline set the result to "Whacky" this would set the internal ext status (whatever it is) to FAILED - when its somewhere between success and unstable.